File: backend/models/resource.js
module.exports = (sequelize, DataTypes) => {
  return sequelize.define('Resource', {
    resourceId: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    courseId: {
      type: DataTypes.UUID,
      allowNull: false,
      references: {
        model: 'Courses',
        key: 'id'
      },
      onDelete: 'CASCADE',
      onUpdate: 'CASCADE'
    },
    teacherId: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'Teachers',
        key: 'id'
      },
      onDelete: 'CASCADE',
      onUpdate: 'CASCADE'
    },
    title: {
      type: DataTypes.STRING,
      allowNull: false
    },
    description: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    type: {
      type: DataTypes.STRING,
      allowNull: false
    },
    friendlyType: {
      type: DataTypes.STRING,
      allowNull: true

    },
    fileName: {
      type: DataTypes.STRING,
      allowNull: false
    },
    originalName: {
      type: DataTypes.STRING,
      allowNull: false
    },
    filePath: {
      type: DataTypes.STRING,
      allowNull: false
    },
    fileType: {
      type: DataTypes.STRING,
      allowNull: false
    },
    fileSize: {
      type: DataTypes.INTEGER,
      allowNull: false
    },
    uploadDate: {
      type: DataTypes.DATE,
      allowNull: false
    },
    url:{
      type: DataTypes.STRING,
      allowNull:true
    },
    content:{
      type: DataTypes.TEXT,
      allowNull:true
    },
    tags: {
      type: DataTypes.ARRAY(DataTypes.STRING), // or DataTypes.STRING if stored as CSV
      allowNull: true
    }
  });
};

File: app/src/main/java/com/example/lmsapp/ui/data/Resource.kt
package com.example.lmsapp.ui.data.DataClasses

import ResourceType
import com.example.lmsapp.ui.data.UserInfo
import kotlinx.serialization.Serializable

@Serializable
data class Resource(
    val resourceId: String,
    val courseId: String?,
    val teacherId: String?,
    val title: String,
    val description: String?,
    val type: ResourceType?,
    val friendlyType: String? = null,
    val url: String?, // For video links, file URLs
    val fileName: String?, // For files
    val originalName: String?,
    val filePath: String?,
    val fileSize: Long?,
    val uploadDate: String?,
    val content: String?, // For notes directly stored
    val tags: List<String>?, // For filtering
    val createdAt: String, // Use String for simplicity, parse to LocalDateTime if needed for display
    val uploadedBy: UserInfo?, // Or just userId and fetch user info
    val teacherName: String? = null
)

